test/probes/vision/vision-4-and-below.js

```diff
@@ -0,0 +1,315 @@
+'use strict'
+
+const base = process.cwd()
+const path = require('path')
+
+const helloDotEjs = 'hello.ejs'
+
+const helper = require(path.join(base, 'test/helper'))
+const ao = helper.ao
+const semver = require('semver')
+
+const request = require('request')
+
+// Don't even load hapi in 0.8. Bad stuff will happen.
+const nodeVersion = process.version.slice(1)
+const hasES6 = semver.satisfies(nodeVersion, '> 4')
+const hapiPkg = require('hapi/package.json')
+let hapi
+let vision
+let pkg
+if (semver.satisfies(nodeVersion, '> 0.8')) {
+  if (hasES6 || semver.satisfies(hapiPkg.version, '< 13.6')) {
+    hapi = require('hapi')
+  }
+
+  pkg = require('vision/package.json')
+  if (hasES6 || semver.satisfies(pkg.version, '<= 4.1.1')) {
+    vision = require('vision')
+  }
+}
+
+
+describe('probes.vision ' + pkg.version + ' hapi ' + hapiPkg.version, function () {
+  let emitter
+  let port = 3000
+
+  //
+  // Intercept appoptics messages for analysis
+  //
+  before(function (done) {
+    helper.ao.resetRequestStore()
+    ao.probes.fs.enabled = false
+    emitter = helper.appoptics(done)
+    ao.sampleRate = ao.addon.MAX_SAMPLE_RATE
+    ao.sampleMode = 'always'
+  })
+  after(function (done) {
+    ao.probes.fs.enabled = true
+    emitter.close(done)
+  })
+
+  const check = {
+    'http-entry': function (msg) {
+      msg.should.have.property('Layer', 'nodejs')
+      msg.should.have.property('Label', 'entry')
+    },
+    'http-exit': function (msg) {
+      msg.should.have.property('Layer', 'nodejs')
+      msg.should.have.property('Label', 'exit')
+    },
+    'hapi-entry': function (msg) {
+      msg.should.have.property('Layer', 'hapi')
+      msg.should.have.property('Label', 'entry')
+    },
+    'hapi-exit': function (msg) {
+      msg.should.have.property('Layer', 'hapi')
+      msg.should.have.property('Label', 'exit')
+    },
+    'render-exit': function (msg) {
+      msg.should.have.property('Layer', 'render')
+      msg.should.have.property('Label', 'exit')
+    }
```

Comment:
No, they're not used. In fact, the entire controllerTest section is not used. The disabled and renderTest sections still run. I left them in because 1) I might get around to adding the controller tests to the suite and 2) it's just testing code so no cost to end-users. I probably should have added a comment to that effect.

The reason the test isn't performed now is because it's redundant - the hapi.test.js code executes the same tests (well, with the fixes you noted for the name changes, etc.) I split vision into a separate test because it must be paired with the correct version of hapi, e.g., hapi v17 requires vision v5. 

I separately enhanced testeachversion to handle some basic dependencies, so hapi v17 can be dependent on vision v5 (and vice versa). this allows the matrix test to execute all versions of vision v5 against  the latest version of hapi v17. I chose not to test all versions of hapi against all compatible versions of vision because it seemed like overkill. But because hapi runs the controllerTest it didn't seem critical for vision to run it also - vision needs to execute only the tests that involve vision, i.e., those that specify `views` in the config.